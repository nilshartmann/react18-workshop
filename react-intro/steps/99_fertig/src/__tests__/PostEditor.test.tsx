import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CreatePostPage from "../PostEditor";
import { expect, test } from "vitest";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { ReactNode } from "react";

type WrapperProps = { children: ReactNode };
function Wrapper({ children }: WrapperProps) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 0,
        retry: false,
        refetchOnMount: false,
        refetchOnReconnect: false
      }
    }
  });

  return <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>;
}

test("renders correctly", () => {
  const { asFragment } = render(<CreatePostPage />, { wrapper: Wrapper });
  expect(asFragment()).toMatchSnapshot();

  screen.logTestingPlaygroundURL();
});

test("save button enablement", async () => {
  const user = userEvent.setup();
  render(<CreatePostPage />, { wrapper: Wrapper });

  const titleInput = screen.getByLabelText("Title");
  const bodyInput = screen.getByLabelText("Body");
  const saveButton = screen.getByRole("button", { name: "Save Post" });

  // Save Button should be disabled
  expect(saveButton).toBeDisabled();

  // enter Title...
  await user.type(titleInput, "New Title");

  // should still be disabled
  expect(saveButton).toBeDisabled();

  // enter body
  await user.type(bodyInput, "New Body");

  // ...now the button should be enabled
  expect(saveButton).toBeEnabled();
});
