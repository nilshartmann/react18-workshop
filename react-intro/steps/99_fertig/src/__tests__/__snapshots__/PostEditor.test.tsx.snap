// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="PostEditor__title"
    >
      <h1>
        PostEditor
      </h1>
      <button
        class="small"
      >
        Show Preview
      </button>
    </div>
    <div
      class="Container"
    >
      <div>
        <h1>
          Create Post
        </h1>
        <label>
          Title
          <input
            value=""
          />
        </label>
        <p
          class="error"
        >
          Please enter a title
        </p>
        <label>
          Body
          <textarea />
        </label>
        <p
          class="error"
        >
          Please enter a body
        </p>
        <div>
          <h2>
            Tags
          </h2>
          <div
            class="TagChooser__tags"
          >
            <label
              class="Checkbox"
            >
              <input
                type="checkbox"
              />
              React
            </label>
            <label
              class="Checkbox"
            >
              <input
                type="checkbox"
              />
              TypeScript
            </label>
            <label
              class="Checkbox"
            >
              <input
                type="checkbox"
              />
              JavaScript
            </label>
          </div>
        </div>
        <button
          disabled=""
        >
          Clear
        </button>
      </div>
    </div>
    <div
      class="PostEditor__buttons"
    >
      <button
        disabled=""
      >
        Save Post
      </button>
    </div>
  </div>
</DocumentFragment>
`;
